include/core/sdirk_table.h

```diff
@@ -0,0 +1,93 @@
+#ifndef SDIRK_TABLE_H
+#define SDIRK_TABLE_H
+
+#include <deal.II/lac/full_matrix.h>
+
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+using namespace dealii;
+
+struct SDIRKTable
+{
+  FullMatrix<double>  A;
+  std::vector<double> b;
+  std::vector<double> c;
+};
+
+
+/**
+ * @brief Construct the Butcher tableau (A, b, c) for a given SDIRK method.
+ *
+ * This function returns the Runge-Kutta coefficients for a diagonally implicit
+ * Runge-Kutta (SDIRK) time integration method, such as SDIRK2 (Alexander's
+ * method here) or SDIRK3 (Kennedy-Carpenter method here). These coefficients
+ * define the time-stepping scheme and are used in implicit time integration of
+ * ODEs or PDEs.
+ *
+ * The Butcher tableau is defined as follows:
+ * - A: the matrix of coefficients a_ij
+ * - b: the weights used in the final combination of stages
+ * - c: the time locations of each stage
+ *
+ * @param[in] method_name A string identifying the SDIRK method (e.g. "SDIRK2",
+ * "SDIRK3").
+ *
+ * @return SDIRKTable A structure containing the Butcher tableau (A, b, c) for the requested method.
+ *
+ * @throws std::invalid_argument if the method name is not recognized.
+ */
+
+inline SDIRKTable
+sdirk_table(const std::string &method_name)
+{
+  SDIRKTable table;
+
+  if (method_name == "SDIRK2")
```

Comment:
There is already an ENUM for the time_stepping method (see in the parameters or simulation control for the definition). I would use the same enum and add sdirk2, sdirk 3... etc. To it.
I would also keep a nomenclature where you have sdirkOrderStage or sdirkStageOrder depending on what the literature uses. I think it's sdirkOrderStage so that SDIRK22 or SDIRK33 would mean order 2 with 2 stage and order 3 with 3 stages respectively.

Reply:
Using the enum will prevent you from having to use string for equality